backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala

```diff
@@ -107,16 +109,24 @@ object VeloxBackendSettings extends BackendSettingsApi {
       val filteredRootPaths = distinctRootPaths(rootPaths)
       if (filteredRootPaths.nonEmpty) {
         val resolvedPaths =
-          if (
-            GlutenConfig.getConf.enableHdfsViewfs && filteredRootPaths.head.startsWith("viewfs")
-          ) {
+          if (GlutenConfig.getConf.enableHdfsViewfs) {
+            val viewfsToHdfsCache: mutable.Map[String, String] = mutable.Map.empty
             // Convert the viewfs path to hdfs path.
             filteredRootPaths.map {
-              viewfsPath =>
-                val viewPath = new Path(viewfsPath)
-                val viewFileSystem =
-                  FileSystem.get(viewPath.toUri, serializableHadoopConf.get.value)
-                viewFileSystem.resolvePath(viewPath).toString
+              path =>
+                if (path.startsWith("viewfs")) {
+                  val pathSplit = path.split(Path.SEPARATOR)
```

Comment:
Can we move this code into a new method in the ViewFileSystemUtils object? This would help in keeping the code reusable.

Reply:
done